### PR TITLE
[FEATURE] Add Genesis Grasping Environment for Reinforcement Learning 

### PR DESCRIPTION
**Pull request by yun-long**
## 🚀 Motivation

This PR introduces a grasping environment built on the Genesis simulator, designed for data-driven manipulation research using a Franka Panda arm. The goal is to provide a lightweight simulation environment that supports reinforcement learning. The code base will also later be extended to learning vision-based manipulation control. 

---

### 🧩 Environment (`GraspEnv`)
- Modular Genesis-based scene with:
  - Robot arm (Franka Panda) + graspable object.
  - Configurable action/observation spaces and reward terms.
  - Visual camera option for inspection/debugging.
---

### 🤖 Robot Abstraction (`Manipulator`)
- High-level controller for robot end-effector relative pose.
- An additional inverse kinematics method:
  - `dls_ik`: Damped Least Squares Jacobian-based IK.

---
### 🤖 Reinforcement learning training and evaluation example

https://github.com/user-attachments/assets/00a40aa9-3eeb-477b-bf9b-35e4cf593cf0

## Related Issue

Resolves Genesis-Embodied-AI/Genesis#<your-issue-number>

## Checklist:

- [x] I read the **CONTRIBUTING** document.
- [x] I followed the `Submitting Code Changes` section of **CONTRIBUTING** document.
- [x] I tagged the title correctly (including BUG FIX/FEATURE/MISC/BREAKING)
- [ ] I updated the [documentation](https://github.com/Genesis-Embodied-AI/genesis-doc) accordingly or no change is needed.
- [ ] I tested my changes and added instructions on how to test it for reviewers.

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/manipulation/grasp_env.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+        self.robot.reset(envs_idx)
+
+        # reset object
+        num_reset = len(envs_idx)
+        random_x = torch.rand(num_reset, device=self.device) * 0.4 + 0.2  # 0.2 ~ 0.6
+        random_y = (torch.rand(num_reset, device=self.device) - 0.5) * 0.5  # -0.25 ~ 0.25
+        random_z = torch.ones(num_reset, device=self.device) * 0.025  # 0.15 ~ 0.15
+        random_pos = torch.stack([random_x, random_y, random_z], dim=-1)
+
+        #
```
Comment (YilingQiao) [inline examples/manipulation/grasp_env.py]: can you add comments or delete it.

File: examples/manipulation/grasp_env.py [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+                torch.zeros(num_reset, device=self.device),
+                torch.zeros(num_reset, device=self.device),
+                torch.sin(random_yaw / 2),
+            ],
+            dim=-1,
+        )
+        goal_yaw = transform_quat_by_quat(q_yaw, q_downward)
+
+        self.goal_pose[envs_idx] = torch.cat([random_pos, goal_yaw], dim=-1)
+        #
```
Comment (YilingQiao) [inline examples/manipulation/grasp_env.py]: same

File: uv.lock [hunk 3]
```diff
@@ -0,0 +1,3074 @@
+version = 1
```
Comment (YilingQiao) [inline uv.lock]: is this file uploaded by accident?

**Comment by YilingQiao** [comment]
Thank you so much! This looks great.
Could you please add a README file? Or add a simple documentation? (e.g. https://genesis-world.readthedocs.io/en/latest/user_guide/getting_started/locomotion.html)

File: uv.lock (hunk 3 above)
Comment (Kashu7100) [reply uv.lock]: I think we shouldn't include `uv.lock`?

File: examples/manipulation/grasp_env.py (hunk 1 above)
Comment (yun-long) [reply examples/manipulation/grasp_env.py]: done. deleted all redundant comments symbols without comments

File: uv.lock (hunk 3 above)
Comment (yun-long) [reply uv.lock]: yes. it was a mistake. I didn't notice it. sorry. thanks for pointing it out.

File: examples/manipulation/grasp_env.py [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+        self._robot_entity.set_dofs_force_range(
+            torch.tensor([-87, -87, -87, -87, -12, -12, -12, -100, -100]),
+            torch.tensor([87, 87, 87, 87, 12, 12, 12, 100, 100]),
+        )
+
+    def _init(self):
+        self._arm_dof_dim = self._robot_entity.n_dofs - 2  # total number of arm: joints
+        self._gripper_dim = 2  # number of gripper joints
+
+        #
```
Comment (YilingQiao) [inline examples/manipulation/grasp_env.py]: still some redundant # signs

File: examples/manipulation/grasp_env.py [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
+            )
+            * unit_length
+        )
+        return keypoint_offsets.unsqueeze(0).repeat(batch_size, 1, 1)
+
+    def grasp_and_lift_demo(self, render: bool = False):
+        total_steps = 500
+        goal_pose = self.robot.ee_pose.clone()
+
+        #
```
Comment (YilingQiao) [inline examples/manipulation/grasp_env.py]: same

File: examples/manipulation/grasp_env.py (hunk 4 above)
Comment (yun-long) [reply examples/manipulation/grasp_env.py]: should be good now.

**Comment by yun-long** [comment]
Regarding the documentation, I will work on it after finishing the vision-based example.

File: examples/drone/hover_env.py
Comment (duburcqa) [inline examples/drone/hover_env.py:1]: Avoid unnecessary formatting changes
